kiota_serialization_json/json_parse_node.py

```diff
@@ -215,7 +215,7 @@ def get_bytes_value(self) -> Optional[bytes]:
         Returns:
             bytearray: The bytearray value from the nodes
         """
-        base64_string = str(self._json_node)
+        base64_string = json.dumps(self._json_node)
```

Comment:
this change only affects the byte array deserialization, I'm not sure how this interacts with my previous PR, but looks like a slightly different scope from a first look.